### PR TITLE
EVG-12535: read concern majority

### PR DESCRIPTION
**Pull request by hhoke**

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -681,10 +682,24 @@ func (wc WriteConcern) Resolve() *writeconcern.WriteConcern {
...
+type ReadConcern struct {
+	Level string `yaml:"level"`
+}
+
+func (wc ReadConcern) Resolve() *readconcern.ReadConcern {
+
+	if wc.Level == "majority" {
+		return readconcern.Majority()
+	} else {
+		return readconcern.New()
```
Comment (bsamek) [inline config.go:694]: Is this a valid readconcern to return? I worry about someone changing code elsewhere and then ending up setting `readconcern.New()`.

Comment (hhoke) [reply config.go:694]: Not really sure. I was setting it to readconcern.Available(), but this caused a variety of errors of the form "The readConcern level must be either 'local' or 'majority' in order to upconvert the readConcern level to 'snapshot'". This got rid of those errors.

Comment (bsamek) [reply config.go:694]: I'd suggest either setting a reasonable default, if there is one, or defaulting to majority.

Comment (hhoke) [reply config.go]: @bsamek changed it to default to Majority

**Pull request merged**